systems/fib.asm

```diff
@@ -0,0 +1,32 @@
+section .text
+global fib
+fib:
+	; deal with the base case
+	mov eax, edi
+	cmp edi, 1 
+	jle return
+
+	; move callee saved registers to the stack
+	push rbx
+	push rbp
+
+	; use ebx to store the parameter n
+	mov ebx, eax
+
+	; call fib(n-1)
+	lea edi, [eax - 1]
+	call fib ; now eax contains fib(n-1)
+	mov ebp, eax ; store fib(n-1) in ebp
+
+	; call fib(n-2)
+	lea edi, [ebx - 2]
+	call fib ; now eax contains fib(n-2)
+
+	; store return value in eax
+	add eax, ebp ; add fib(n-1) to fib(n-2) 
+
+	; restore callee saved registers from stack to register
+	pop rbp
+	pop rbx
```

Comment:
Was there a reason you were pushing/popping the 64-bit version of the registers but using the 32-bit versions?

Reply:
when I tried to push/pop a 32 bit register, I got the following "error: instruction not supported in 64-bit mode"
when I tried pushl/popl, I got the following "error: parser: instruction expected"
I could just move the stack pointer and mov 32 bits at a time into their respective locations, but I was just being lazy. xD